lib/sparql.jison

```diff
@@ -266,6 +266,54 @@
     }
     return merged;
   }
+
+  // Return the id of an expression
+  function getIdOfExpression(expression) {
+    return expression.variable ? expression.variable.id : expression.id || expression.expression.id;
```

Comment:
`.id` is implementation-specific; @rubensworks, what would you recommend? N3.js exposes `termToId` if needed (but I don't think it is)

Reply:
I'm not that familiar with this codebase, but it looks like only variables (or variables in expressions) are passed here (but not sure). In that case, I guess we could just use `.value` instead of `.id`.

If non-variable terms are also applicable here, then [`termToString`](https://github.com/rubensworks/rdf-string.js#term-to-string) might be the safest solution here.

Reply:
The variables the parser gives me are in this shape: `Variable { id: '?x' }` . There is no value field availble (it's undefined if I print it), which makes the .value and termToString option not possible).
Expressions are of the following form (as I get them from the parser):  `{ expression: Variable { id: '?x' } }` . When using `termToId` of N3.js, it shows the following error: `Unexpected termType: undefined` . I'm not entirely sure what to do now.

Reply:
`.value` on those terms should be possible though. It's not because they are not printed when calling `console.log(term)`, that they are not there. Not all fields in JS objects are necessarily printed.

An expression is not an RDF/JS term AFAIK, so a crash when calling term-operations on it is expected. So if you want to determine some unique id for those expressions, you'll have to look for a workaround, possibly by checking what this expression is defined over.